core/TypeDrivenAutocorrect.cc

```diff
@@ -18,6 +18,15 @@ void TypeDrivenAutocorrect::maybeAutocorrect(const GlobalState &gs, ErrorBuilder
         if (!withoutNil.isBottom() &&
             Types::isSubTypeUnderConstraint(gs, constr, withoutNil, expectedType, UntypedMode::AlwaysCompatible)) {
             e.replaceWith("Wrap in `T.must`", loc, "T.must({})", loc.source(gs).value());
+        } else if (Types::isSubTypeUnderConstraint(gs, constr, expectedType, Types::Boolean(),
+                                                   UntypedMode::AlwaysCompatible)) {
+            if (core::isa_type<ClassType>(actualType)) {
+                auto classSymbol = core::cast_type_nonnull<ClassType>(actualType).symbol;
+                if (classSymbol.exists() && classSymbol.data(gs)->owner == core::Symbols::root() &&
+                    classSymbol.data(gs)->name == core::Names::Constants::Boolean()) {
```

Comment:
Why only `Boolean`? It seems like a lot of other cases could benefit from such an autocorrect?

Reply:
Anything that is a BasicObject could potentially be autocorrected here since `!` is defined: https://ruby-doc.org/core-2.5.1/BasicObject.html#method-i-21

Reply:
I thought so too, but then I couldn’t come up with a compelling test case, so I erred on the cautious side and didn’t make it general. Do you have a good test case that it should apply to?

Reply:
Specifically, given how few things are falsy in Ruby, all the examples I came up with seemed contrived. Like Integer to T::Boolean autocorrect seems bad—that almost certainly indicates a bug in your code, because Integer is always truthy.

Reply:
It depends how you see the autocorrect:

* If it's a tool used to mass-port a project to correct typing then indeed Sorbet should not suggest this fix.
* If it's a tool that suggests solutions to local problems that can be hand-picked by the developer, then it makes sense to suggest it.

I personally see it as the later rather than the former so for me it would make sense to always suggest it and let the developer decide if they want to apply it or not.

Reply:
I think that for the time being I'll ship the minimal thing. I'm open to expanding this in the future.